#!/usr/bin/env python

PACKAGE = "mecanum_wheels"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

integration_method_enum = gen.enum([gen.const("Euler", str_t, "EULER", "Euler Integration"),
                                    gen.const("RungeKutta", str_t, "RUNGEKUTTA", "RungeKutta integration")],
                                    "Enum of formats")

gen.add("integration_method", str_t, 0, "integration method", default="RUNGEKUTTA", edit_method=integration_method_enum)

gen.add("half_length", double_t, 0, "half_length set", default=0.2, min=0, max=0.5)
gen.add("half_width", double_t, 0, "half_width set", default=0.169, min=0, max=0.5)
gen.add("wheel_radius", double_t, 0, "wheel_radius set", default=0.075, min=0, max=0.2)
gen.add("encoder_resolution", double_t, 0, "encoder_resolution set", default=42, min=35, max=50)
gen.add("reset_to_initial_position", bool_t, 0, "To be used to change bag.", default=False)

exit(gen.generate(PACKAGE, "odometry", "parameters"))